### ISSUE TITLE
v2: Use `tools/copyright` and add our own COPYRIGHT headers

### ISSUE THREAD

**Issue by bwplotka** [grpc-ecosystem/go-grpc-middleware#389]
Example copyright we could use:

```
// Copyright (c) The go-grpc-middleware Authors.
// Licensed under the Apache License 2.0.
```

I would recommend using https://github.com/efficientgo/tools#githubcomefficientgotoolscopyright -> it's easy to setup, we use it slowly everywhere (:

**Comment by bwplotka** [comment]
cc @yashrsharma44 @MalloZup if you have time 🤗

**Comment by yashrsharma44** [comment]
Can take this issue if @MalloZup isn't taking 😁

**Comment by MalloZup** [comment]
@yashrsharma44  go ahead.  :sunflower: 

I want to carry over the thanos implementation of rate-limit  within the middleware :racehorse:

**Comment by yashrsharma44** [comment]
Added a PR for the same, let me know if the changes make sense :)

**Comment by yashrsharma44** [comment]
Resolved by #420
